src/System.Runtime/tests/System/TypeTests.netcoreapp.cs

```diff
@@ -36,5 +36,42 @@ public void IsSZArray(Type type, bool expected)
         {
             Assert.Equal(expected, type.IsSZArray);
         }
+
+        private static IEnumerable<object[]> VariableBoundArrayOrNotTypes()
+        {
+            // Not arrays
+            yield return new object[] { typeof(void), false };
+            yield return new object[] { typeof(int), false };
+            yield return new object[] { typeof(int[]).MakeByRefType(), false };
+            yield return new object[] { typeof(TypeTests), false };
+            yield return new object[] { Type.GetType("System.Int32"), false };
+            yield return new object[] { typeof(Outside<int>.Inside<string>), false };
+
+            // Arrays, but SZ arrays
+            yield return new object[] { Type.GetType("System.Int32[]"), false };
+            yield return new object[] { typeof(int[]), false };
```

Comment:
>For the MakeArrayType(1) case, please enclose it in a guard 

Actually, let me amend this one. Instead of skipping this on .NET Native, make it an a rank-1 md array of an open type, like this:

  typeof(Outer<>).GetTypeInfo().GenericTypeParameters[0].MakeArrayType(1)

This will work even on .NET Native.

This is the key test case for this api so we don't really want to skip it...